### ISSUE TITLE
nxos_l2_interfaces does not remove native_vlan if previous set

### ISSUE THREAD

**Issue by tweippert** [ansible-collections/cisco.nxos#21]
##### SUMMARY

On a Trunk Port with formerly native_vlan setting, a new run without_native vlan and replace state does not remove the native_vlan configuration and no changed state.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

nxos_l2_interfaces

##### ANSIBLE VERSION

```paste below
ansible 2.9.6
  config file = /home/weippertt/.ansible.cfg
  configured module search path = ['/home/weippertt/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/weippertt/ansible-2.9.6/lib/python3.6/site-packages/ansible
  executable location = /home/weippertt/ansible-2.9.6/bin/ansible
  python version = 3.6.8 (default, Jun 11 2019, 15:15:01) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]

```

##### CONFIGURATION

```paste below
HOST_KEY_CHECKING(/home/weippertt/.ansible.cfg) = False
PERSISTENT_COMMAND_TIMEOUT(/home/weippertt/.ansible.cfg) = 300
PERSISTENT_CONNECT_RETRY_TIMEOUT(/home/weippertt/.ansible.cfg) = 300
PERSISTENT_CONNECT_TIMEOUT(/home/weippertt/.ansible.cfg) = 300
```

##### OS / ENVIRONMENT

Ansible running on Red Hat Enterprise Linux Server release 7.6 (Maipo)
Cisco Device is:
cisco Nexus 56128P Chassis ("Nexus 56128P Supervisor")
system: version 7.3(3)N1(1)
bootflash:///n6000-uk9.7.3.3.N1.1.bin

##### STEPS TO REPRODUCE

Run the tasks once with pUntagged set to an vlan id -> native_vlan was set
Run the tasks again with pUntagged set to NONE -> native_vlan was null but not removed.

```yaml
- name: Configure Trunk Port (L2)
  nxos_l2_interfaces:
    config:
      - name: "{{ item.value.pName }}"
        trunk:
          allowed_vlans: "{{ item.value.pTagged | join(',') }}"
    state: replaced
  when: item.value.pMode == 'TAGGED' and item.value.pUntagged == 'NONE'
  register: port_trunk
  notify: save_nxos

- name: Configure Native-VLAN Trunk Port (L2)
  nxos_l2_interfaces:
    config:
      - name: "{{ item.value.pName }}"
        trunk:
          allowed_vlans: "{{ item.value.pTagged | join(',') }}"
          native_vlan: "{{ item.value.pUntagged }}"
    state: replaced
  when: item.value.pMode == 'TAGGED' and item.value.pUntagged != 'NONE'
  register: port_trunk_native
  notify: save_nxos

```

##### EXPECTED RESULTS

Remove native_vlan from configured port.

##### ACTUAL RESULTS

NAtive VLan was left as -is and task has not changed

```paste below
    "invocation": {
        "module_args": {
            "config": [
                {
                    "access": null,
                    "name": "Ethernet1/52",
                    "trunk": {
                        "allowed_vlans": "10,11,998,999",
                        "native_vlan": null
                    }
                }
            ],
            "state": "replaced"
        }
    }

```

**Comment by NilashishC** [comment]
Hi @tweippert. Could you please confirm if https://github.com/ansible-collections/cisco.nxos/pull/55 fixes the issue for you? Thanks!

**Comment by tweippert** [comment]
Hi @NilashishC, also this fix seems to work as expected, native vlan gets removed properly if not configured.

Is there a change that this get backported/incorporated in 2.9.X Versions?

**Comment by NilashishC** [comment]
@tweippert thank you for confirming. Yes, this fix will be backported and released with the next 2.9 release.
